azurerm/internal/services/machinelearning/resource_arm_machine_learning_workspace.go

```diff
@@ -0,0 +1,425 @@
+				// TODO -- remove when issue https://github.com/Azure/azure-rest-api-specs/issues/8323 is addressed
+				DiffSuppressFunc: suppress.CaseDifference,
+			},
+
+			"identity": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(machinelearningservices.SystemAssigned),
+							}, false),
+						},
+						"principal_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"tenant_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"container_registry_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				// TODO -- use the custom validation function of container registry
+				ValidateFunc: azure.ValidateResourceID,
+				// TODO -- remove when issue https://github.com/Azure/azure-rest-api-specs/issues/8323 is addressed
+				DiffSuppressFunc: suppress.CaseDifference,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"friendly_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"sku_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"Basic",
+					"Enterprise",
+				}, true),
+				Default: "Basic",
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceArmMachineLearningWorkspaceCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).MachineLearning.WorkspacesClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+
+	existing, err := client.Get(ctx, resGroup, name)
+	if err != nil {
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("Error checking for existing AML Workspace %q (Resource Group %q): %s", name, resGroup, err)
+		}
+	}
+	if existing.ID != nil && *existing.ID != "" {
+		return tf.ImportAsExistsError("azurerm_machine_learning_workspace", *existing.ID)
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	storageAccountId := d.Get("storage_account_id").(string)
+	keyVaultId := d.Get("key_vault_id").(string)
+	applicationInsightsId := d.Get("application_insights_id").(string)
+	skuName := d.Get("sku_name").(string)
+
+	identityRaw := d.Get("identity").([]interface{})
+	identity := expandArmMachineLearningWorkspaceIdentity(identityRaw)
+
+	t := d.Get("tags").(map[string]interface{})
+
+	workspace := machinelearningservices.Workspace{
+		Name:     &name,
+		Location: &location,
+		Tags:     tags.Expand(t),
+		Sku:      &machinelearningservices.Sku{Name: utils.String(skuName)},
+		Identity: identity,
```

Comment:
Given that this is always required, i think we could
```suggestion
		Identity: utils.String(d.Get("identity.0.type").(string)),
```

Reply:
Well, identity is a block, therefore I bring the definition of `identity` here in the struct initialization